test/Spec/BeneficiariesFile.hs

```diff
@@ -1,35 +1,156 @@
    in didTruncate === shouldTruncate
 
+prop_Parse :: Beneficiaries -> Property
+prop_Parse (Beneficiaries (bens, config)) = either (error . unpack) (const $ property True) $ parseContent config bens
+
+-- TODO Generate [Beneficiary] directly, don't depend on parsing
+prop_Print :: Beneficiaries -> Property
+prop_Print (Beneficiaries (bens, config)) = either (error . unpack) (const $ property True) $ parseContent config bens >>= prettyContent config
+
+prop_Roundtrip1 :: Beneficiaries -> Property
+prop_Roundtrip1 (Beneficiaries (bens, config)) =
+  -- We can't check for equality immediately from the incoming Text, as information can be lost during parsing.
+  -- Thus we initially parse and then print, and then do the roundtrip, as parse >=> print should be idempotent
+  let printed = either (error . unpack) id $ parseContent config bens >>= prettyContent config
+      roundTrip = parseContent config >=> prettyContent config
+   in roundTrip printed === Right printed
+
+-- TODO Generate [Beneficiary] directly
+prop_Roundtrip2 :: Beneficiaries -> Property
+prop_Roundtrip2 (Beneficiaries (bens, config)) =
+  let parsed = either (error . unpack) id $ parseContent config bens
+      roundTrip = prettyContent config >=> parseContent config
+   in roundTrip parsed === Right parsed
+
+newtype TokenAmount = TokenAmount {unTokenAmount :: Scientific}
+  deriving stock (Show)
+
+newtype Beneficiaries = Beneficiaries (Text, Config) deriving stock (Show)
+
+instance Arbitrary TokenAmount where
+  arbitrary = do
+    Positive (Large x) <- arbitrary @(Positive (Large Int))
+    Small scale <- arbitrary @(Small Int)
+    pure . TokenAmount $ normalize $ scientific (toInteger x) scale
+
+instance Arbitrary Beneficiaries where
+  arbitrary = do
+    usePK <- arbitrary @Bool
+    Positive dp <- arbitrary @(Positive Integer)
+
+    dropAmount' <- fmap unTokenAmount <$> arbitrary @(Maybe TokenAmount)
+    assetClass' <- (defaultConfig.assetClass <*) <$> arbitrary @(Maybe ())
+    let config =
+          defaultConfig
+            { usePubKeys = usePK
+            , decimalPlaces = dp
+            , truncate = True
+            , dropAmount = dropAmount'
+            , assetClass = assetClass'
+            }
+    beneficiaries <- listOf1 $ beneficiary usePK (isNothing assetClass') (isNothing dropAmount')
+    pure $ Beneficiaries (unlines beneficiaries, config)
+    where
+      beneficiary :: Bool -> Bool -> Bool -> Gen Text
+      beneficiary usePubKey addAssetClass addTokenAmount = do
+        let destination = if usePubKey then pkhs else addresses
+            tokenAmount = do
+              amt <- arbitrary @TokenAmount
+              pure $ (pack . formatScientific Fixed Nothing . unTokenAmount $ amt) <$ guard addTokenAmount
+            currencySymbol = do
+              symbol <- currencySymbols
+              maybeTokenName <- liftArbitrary @Maybe tokenNames
+              pure $ maybe symbol ((symbol <> ".") <>) maybeTokenName <$ guard addAssetClass
```

Comment:
If we're guarding by `addAssetClass`, then we may as well put that guard at the top of the computation.
that'll let to make this line a little more readable

Reply:
Moved the guarding to the array of parts, to be the same style as is done in `prettyBeneficiary`